package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,270 @@
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, atomgroup, select='all', align=False, mean=None,
+                 n_components=None, **kwargs):
+        """
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup
+            AtomGroup to be used for PCA.
+        select: string, optional
+            A valid selection statement for choosing a subset of atoms from
+            the atomgroup.
+        align: boolean, optional
+            If True, the trajectory will be aligned to a reference
+            structure.
+        mean: MDAnalysis atomgroup, option
+            An optional reference structure to be used as the mean of the
+            covariance matrix.
+        n_components : int, optional
+            The number of principal components to be saved, default saves
+            all principal components, Default: None
+        start : int, optional
+            First frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        stop : int, optional
+            Last frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: None
+        """
+        super(PCA, self).__init__(atomgroup.universe.trajectory,
+                                  **kwargs)
+        self._u = atomgroup.universe
+        if self._quiet:
+            logging.disable(logging.WARN)
+        # for transform function
+        self.align = align
+        # access 0th index
+        self._u.trajectory[0]
+        # reference will be 0th index
+        self._reference = self._u.select_atoms(select)
+        self._atoms = self._u.select_atoms(select)
+        self.n_components = n_components
+        self._n_atoms = self._atoms.n_atoms
+        self._calculated = False
+        if mean is None:
+            logger.warn('In order to demean to generate the covariance matrix\n'
+                        + 'the frames have to be iterated over twice. To avoid\n'
+                        'this slowdown, provide an atomgroup for demeaning.')
+            self.mean = np.zeros(642)
+            self._calc_mean = True
+        else:
+            self.mean = mean.positions
+            self._calc_mean = False
+
+    def _prepare(self):
+        n_dim = self._n_atoms * 3
+        self.cov = np.zeros((n_dim, n_dim))
+        self._ref_atom_positions = self._reference.positions
+        self._ref_cog = self._reference.center_of_geometry()
+        self._ref_atom_positions -= self._ref_cog
+
+        if self._calc_mean:
+            for i, ts in enumerate(self._u.trajectory[self.start:self.stop:self.step]):
```

Comment:
this should also be reported with a progress meter. Use the quiet parameter to decide if it should be a verbose progress meter or not.